### ISSUE TITLE
Check Wazuh Agent compatibility with new version Red Hat Enterprise Linux 9.3

### ISSUE THREAD

**Issue by jotacarma90** [wazuh/wazuh#20387]
# Description
Hello team, this issue is to check the full compatibility of Wazuh on the newfound version of Red Hat Enterprise Linux 9.3 operating system.

OSs checks issue: https://github.com/wazuh/wazuh/issues/20373

For this, it is necessary to perform the following tests to check that everything works as expected:

- [x] Agent and server (if possible) instalations.
- [x] Enrollment and connectivity with the manager.
- [x] WPK  upgrade.
- [x] FIM: Realtime and whodata engines (if available).
- [x] SCA: Policy support.
- [x] Vulnerability Detector: Vulnerability support.
- [x] Syscollector: Complete inventory.
- [x] Active Response.
- [x] Log capture (Eventchannel for Windows and ULS for macOS).

**Comment by ncvicchi** [comment]
# Testing

###   :green_circle:  Agent and server, (enrollment and connectivity with the manager)

Agent installed at the RHEL 9.3 leap machine by following [this guide](https://documentation.wazuh.com/current/installation-guide/wazuh-server/index.html).
Manager was implemented by running an the OVA 4.6.0 release.

<details><summary>Agent log</summary>

```console
[root@rhel93 wazuh]# cat /var/ossec/logs/ossec.log
2023/11/26 20:52:21 wazuh-execd: INFO: Started (pid: 9514).
2023/11/26 20:52:22 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2023/11/26 20:52:22 wazuh-agentd: INFO: Using notify time: 10 and max time to reconnect: 60
2023/11/26 20:52:22 wazuh-agentd: INFO: Version detected -> Linux |rhel93 |5.14.0-362.8.1.el9_3.x86_64 |#1 SMP PREEMPT_DYNAMIC Tue Oct 3 11:12:36 EDT 2023 |x86_64 [Red Hat Enterprise Linux|rhel: 9.3 (Plow)] - Wazuh v4.6.0
2023/11/26 20:52:22 wazuh-agentd: INFO: Started (pid: 9526).
2023/11/26 20:52:22 wazuh-agentd: INFO: Requesting a key from server: 192.168.0.153
2023/11/26 20:52:22 wazuh-agentd: INFO: No authentication password provided
2023/11/26 20:52:22 wazuh-agentd: INFO: Using agent name as: rhel93
2023/11/26 20:52:22 wazuh-agentd: INFO: Waiting for server reply
2023/11/26 20:52:22 wazuh-agentd: INFO: Valid key received
2023/11/26 20:52:22 wazuh-agentd: INFO: Waiting 20 seconds before server connection
2023/11/26 20:52:23 wazuh-syscheckd: INFO: Started (pid: 9540).
...
2023/11/26 20:52:42 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2023/11/26 20:52:42 wazuh-agentd: INFO: Using AES as encryption method.
2023/11/26 20:52:42 wazuh-agentd: INFO: Trying to connect to server ([192.168.0.153]:1514/tcp).
2023/11/26 20:52:42 wazuh-agentd: INFO: (4102): Connected to the server ([192.168.0.153]:1514/tcp).
2023/11/26 20:52:42 wazuh-agentd: INFO: Agent is restarting due to shared configuration changes.

```

</details>

<details><summary>wazuh-control status before reboot</summary>

```console

[root@rhel93 wazuh]# /var/ossec/bin/wazuh-control status
wazuh-modulesd is running...
wazuh-logcollector is running...
wazuh-syscheckd is running...
wazuh-agentd is running...
wazuh-execd is running...
[root@rhel93 wazuh]#


```
</details>

<details><summary>wazuh-control status after reboot</summary>

```console

[root@rhel93 wazuh]# /var/ossec/bin/wazuh-control status
wazuh-modulesd is running...
wazuh-logcollector is running...
wazuh-syscheckd is running...
wazuh-agentd is running...
wazuh-execd is running...


```

</details>

<details><summary>Manager log</summary>

```

2023/11/26 23:52:22 wazuh-authd: INFO: New connection from 192.168.0.16
2023/11/26 23:52:22 wazuh-authd: INFO: Received request for a new agent (rhel93) from: 192.168.0.16
2023/11/26 23:52:22 wazuh-authd: INFO: Agent key generated for 'rhel93' (requested by any)
2023/11/26 23:52:25 wazuh-remoted: INFO: (1409): Authentication file changed. Updating.
2023/11/26 23:52:25 wazuh-remoted: INFO: (1410): Reading authentication keys file.

```
</details>

###  :green_circle: FIM scheduled

Using default configuration (just changed frequency to 60 seconds):

<details><summary>Results</summary>

```console

** Alert 1701045664.19527: - ossec,syscheck,syscheck_entry_added,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Nov 27 00:41:04 (rhel93) any->syscheck
Rule: 554 (level 5) -> 'File added to the system.'
File '/etc/testchesdule' added
Mode: scheduled

Attributes:
 - Size: 0
 - Permissions: rw-r--r--
 - Date: Mon Nov 27 00:40:34 2023
 - Inode: 34724752
 - User: root (0)
 - Group: root (0)
 - MD5: d41d8cd98f00b204e9800998ecf8427e
 - SHA1: da39a3ee5e6b4b0d3255bfef95601890afd80709
 - SHA256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855

** Alert 1701045728.20208: - ossec,syscheck,syscheck_entry_deleted,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Nov 27 00:42:08 (rhel93) any->syscheck
Rule: 553 (level 7) -> 'File deleted.'
File '/etc/testchesdule' deleted
Mode: scheduled

Attributes:
 - Size: 0
 - Permissions: rw-r--r--
 - Date: Mon Nov 27 00:40:34 2023
 - Inode: 34724752
 - User: root (0)
 - Group: root (0)
 - MD5: d41d8cd98f00b204e9800998ecf8427e
 - SHA1: da39a3ee5e6b4b0d3255bfef95601890afd80709
 - SHA256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855

```
  
</details>

###  :green_circle: FIM: Real-time

Configuration:
```
    <directories realtime="yes">/testrealtime</directories>
```

<details><summary>Results</summary>

```
[root@rhel93 wazuh]# touch /testrealtime/file1
[root@rhel93 wazuh]# nano /testrealtime/file1
[root@rhel93 wazuh]# rm /testrealtime/file1
rm: ¿borrar el fichero regular '/testrealtime/file1'? (s/n) s

```

```
** Alert 1701045860.21525: - ossec,syscheck,syscheck_entry_added,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Nov 27 00:44:20 (rhel93) any->syscheck
Rule: 554 (level 5) -> 'File added to the system.'
File '/testrealtime/file1' added
Mode: realtime

Attributes:
 - Size: 0
 - Permissions: rw-r--r--
 - Date: Mon Nov 27 00:44:20 2023
 - Inode: 52246751
 - User: root (0)
 - Group: root (0)
 - MD5: d41d8cd98f00b204e9800998ecf8427e
 - SHA1: da39a3ee5e6b4b0d3255bfef95601890afd80709
 - SHA256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855

** Alert 1701045869.22207: - ossec,syscheck,syscheck_entry_modified,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Nov 27 00:44:29 (rhel93) any->syscheck
Rule: 550 (level 7) -> 'Integrity checksum changed.'
File '/testrealtime/file1' modified
Mode: realtime
Changed attributes: size,mtime,md5,sha1,sha256
Size changed from '0' to '5'
Old modification time was: '1701045860', now it is '1701045869'
Old md5sum was: 'd41d8cd98f00b204e9800998ecf8427e'
New md5sum is : 'd8e8fca2dc0f896fd7cb4cb0031ba249'
Old sha1sum was: 'da39a3ee5e6b4b0d3255bfef95601890afd80709'
New sha1sum is : '4e1243bd22c66e76c2ba9eddc1f91394e57f9f83'
Old sha256sum was: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
New sha256sum is : 'f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2'

Attributes:
 - Size: 5
 - Permissions: rw-r--r--
 - Date: Mon Nov 27 00:44:29 2023
 - Inode: 52246751
 - User: root (0)
 - Group: root (0)
 - MD5: d8e8fca2dc0f896fd7cb4cb0031ba249
 - SHA1: 4e1243bd22c66e76c2ba9eddc1f91394e57f9f83
 - SHA256: f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2

** Alert 1701045880.23431: - ossec,syscheck,syscheck_entry_deleted,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Nov 27 00:44:40 (rhel93) any->syscheck
Rule: 553 (level 7) -> 'File deleted.'
File '/testrealtime/file1' deleted
Mode: realtime

Attributes:
 - Size: 5
 - Permissions: rw-r--r--
 - Date: Mon Nov 27 00:44:29 2023
 - Inode: 52246751
 - User: root (0)
 - Group: root (0)
 - MD5: d8e8fca2dc0f896fd7cb4cb0031ba249
 - SHA1: 4e1243bd22c66e76c2ba9eddc1f91394e57f9f83
 - SHA256: f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2


```

</details>

###  :green_circle: FIM whodata

Configuration:
```
    <directories whodata="yes">/testwhodata</directories>
```

<details><summary>Results</summary>

```
[root@rhel93 wazuh]# touch /testwhodata/test.txt
[root@rhel93 wazuh]# nano /testwhodata/test.txt
[root@rhel93 wazuh]# rm /testwhodata/test.txt

```

```

** Alert 1701046045.30788: - ossec,syscheck,syscheck_entry_added,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Nov 27 00:47:25 (rhel93) any->syscheck
Rule: 554 (level 5) -> 'File added to the system.'
File '/testwhodata/test.txt' added
Mode: whodata

Attributes:
 - Size: 0
 - Permissions: rw-r--r--
 - Date: Mon Nov 27 00:47:25 2023
 - Inode: 52246753
 - User: root (0)
 - Group: root (0)
 - MD5: d41d8cd98f00b204e9800998ecf8427e
 - SHA1: da39a3ee5e6b4b0d3255bfef95601890afd80709
 - SHA256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
 - (Audit) User name: root
 - (Audit) Audit name: wazuh
 - (Audit) Effective name: root
 - (Audit) Group name: root
 - (Audit) Process id: 14367
 - (Audit) Process name: /usr/bin/touch
 - (Audit) Process cwd: /home/wazuh
 - (Audit) Parent process name: /usr/bin/bash
 - (Audit) Parent process id: 4772
 - (Audit) Parent process cwd: /home/wazuh

** Alert 1701046053.31816: - ossec,syscheck,syscheck_entry_modified,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Nov 27 00:47:33 (rhel93) any->syscheck
Rule: 550 (level 7) -> 'Integrity checksum changed.'
File '/testwhodata/test.txt' modified
Mode: whodata
Changed attributes: size,mtime,md5,sha1,sha256
Size changed from '0' to '5'
Old modification time was: '1701046045', now it is '1701046053'
Old md5sum was: 'd41d8cd98f00b204e9800998ecf8427e'
New md5sum is : 'd8e8fca2dc0f896fd7cb4cb0031ba249'
Old sha1sum was: 'da39a3ee5e6b4b0d3255bfef95601890afd80709'
New sha1sum is : '4e1243bd22c66e76c2ba9eddc1f91394e57f9f83'
Old sha256sum was: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
New sha256sum is : 'f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2'

Attributes:
 - Size: 5
 - Permissions: rw-r--r--
 - Date: Mon Nov 27 00:47:33 2023
 - Inode: 52246753
 - User: root (0)
 - Group: root (0)
 - MD5: d8e8fca2dc0f896fd7cb4cb0031ba249
 - SHA1: 4e1243bd22c66e76c2ba9eddc1f91394e57f9f83
 - SHA256: f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2
 - (Audit) User name: root
 - (Audit) Audit name: wazuh
 - (Audit) Effective name: root
 - (Audit) Group name: root
 - (Audit) Process id: 14532
 - (Audit) Process name: /usr/bin/nano
 - (Audit) Process cwd: /home/wazuh
 - (Audit) Parent process name: /usr/bin/bash
 - (Audit) Parent process id: 4772
 - (Audit) Parent process cwd: /home/wazuh

** Alert 1701046061.33385: - ossec,syscheck,syscheck_entry_deleted,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Nov 27 00:47:41 (rhel93) any->syscheck
Rule: 553 (level 7) -> 'File deleted.'
File '/testwhodata/test.txt' deleted
Mode: whodata

Attributes:
 - Size: 5
 - Permissions: rw-r--r--
 - Date: Mon Nov 27 00:47:33 2023
 - Inode: 52246753
 - User: root (0)
 - Group: root (0)
 - MD5: d8e8fca2dc0f896fd7cb4cb0031ba249
 - SHA1: 4e1243bd22c66e76c2ba9eddc1f91394e57f9f83
 - SHA256: f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2
 - (Audit) User name: root
 - (Audit) Audit name: wazuh
 - (Audit) Effective name: root
 - (Audit) Group name: root
 - (Audit) Process id: 14907
 - (Audit) Process name: /usr/bin/rm
 - (Audit) Process cwd: /home/wazuh
 - (Audit) Parent process name: /usr/bin/bash
 - (Audit) Parent process id: 4772
 - (Audit) Parent process cwd: /home/wazuh



```

</details>

### :green_circle: SCA support


Configuration

SCA is officially supported on RHEL 9. Default configuration was used.

<details><summary>Agent log</summary>

```console

2023/11/26 21:47:14 sca: INFO: Module started.
2023/11/26 21:47:14 sca: INFO: Loaded policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'

2023/11/26 21:47:14 sca: INFO: Starting Security Configuration Assessment scan.
2023/11/26 21:47:14 sca: INFO: Starting evaluation of policy: '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2023/11/26 21:47:18 sca: INFO: Evaluation finished for policy '/var/ossec/ruleset/sca/cis_rhel9_linux.yml'
2023/11/26 21:47:18 sca: INFO: Security Configuration Assessment scan finished. Duration: 4 seconds.

```

</details>


<details><summary>Results</summary>

![image](https://github.com/wazuh/wazuh/assets/69121070/1a7005b1-ee3f-4f23-9817-2aa2cba82279)

![image](https://github.com/wazuh/wazuh/assets/69121070/dbfc4e90-f327-4709-b748-fe8f804150c4)

</details>

### :green_circle: Syscollector

Configuration by default

<details><summary>Agent log</summary>

```
2023/11/26 21:47:14 wazuh-modulesd:syscollector: INFO: Module started.
2023/11/26 21:47:14 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2023/11/26 21:47:14 wazuh-modulesd:syscollector: INFO: Evaluation finished.

```

</details>

<details><summary>Manager log</summary>

```
2023/09/26 21:49:57 wazuh-modulesd:syscollector: INFO: Module started.
2023/09/26 21:49:57 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2023/09/26 21:50:32 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.
2023/09/26 21:50:43 wazuh-modulesd:syscollector: INFO: Evaluation finished.
```

</details>

<details><summary>Inventory</summary>

![image](https://github.com/wazuh/wazuh/assets/69121070/4bec302a-3201-431d-bb93-fd1f94f9e244)

![image](https://github.com/wazuh/wazuh/assets/69121070/cec9fb63-fa47-4f3b-9634-7ad43eab8e5a)

</details>

###  :green_circle: Vulnerability detector

RHEL 9 is officially supported in [Version 4.6.0](https://documentation.wazuh.com/current/user-manual/capabilities/vulnerability-detection/how-it-works.html#vuln-det-compatibility-matrix)

Both vulnerability detection module as well as RHEL vulnerability detection were enabled at the manager's ossec.log

```console
  <vulnerability-detector>
    <enabled>yes</enabled>
    <interval>5m</interval>
    <min_full_scan_interval>6h</min_full_scan_interval>
    <run_on_start>yes</run_on_start>
...
    <!-- RedHat OS vulnerabilities -->
    <provider name="redhat">
      <enabled>yes</enabled>
      <os>5</os>
      <os>6</os>
      <os>7</os>
      <os>8</os>
      <os>9</os>
      <update_interval>1h</update_interval>
    </provider>
```

<details><summary>Results</summary>

```console

2023/11/27 01:20:35 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'Red Hat Enterprise Linux 5' feed finished successfully.
2023/11/27 01:20:35 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'Red Hat Enterprise Linux 6' database update.
2023/11/27 01:20:47 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'Red Hat Enterprise Linux 6' feed finished successfully.
2023/11/27 01:20:47 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'Red Hat Enterprise Linux 7' database update.
2023/11/27 01:21:00 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'Red Hat Enterprise Linux 7' feed finished successfully.
2023/11/27 01:21:00 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'Red Hat Enterprise Linux 8' database update.
2023/11/27 01:21:14 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'Red Hat Enterprise Linux 8' feed finished successfully.
2023/11/27 01:21:14 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'Red Hat Enterprise Linux 9' database update.
2023/11/27 01:21:20 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'Red Hat Enterprise Linux 9' feed finished successfully.
2023/11/27 01:21:20 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'JSON Red Hat Enterprise Linux' database update.
2023/11/27 01:22:27 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'JSON Red Hat Enterprise Linux' feed finished successfully.
2023/11/27 01:22:27 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'National Vulnerability Database' database update.
2023/11/27 01:26:13 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'National Vulnerability Database' feed finished successfully.
2023/11/27 01:26:13 wazuh-modulesd:vulnerability-detector: INFO: (5400): Starting 'Microsoft Security Update' database update.
2023/11/27 01:26:19 wazuh-modulesd:vulnerability-detector: INFO: (5430): The update of the 'Microsoft Security Update' feed finished successfully.
2023/11/27 01:26:19 wazuh-modulesd:vulnerability-detector: INFO: (5431): Starting vulnerability scan.
2023/11/27 01:26:19 wazuh-modulesd:vulnerability-detector: WARNING: (5575): Unavailable vulnerability data for the agent '000' OS. Skipping it.
2023/11/27 01:26:19 wazuh-modulesd:vulnerability-detector: INFO: (5450): Analyzing agent '003' vulnerabilities.
2023/11/27 01:29:41 wazuh-modulesd:vulnerability-detector: INFO: (5471): Finished vulnerability assessment for agent '003'
2023/11/27 01:29:41 wazuh-modulesd:vulnerability-detector: INFO: (5472): Vulnerability scan finished.

```

![image](https://github.com/wazuh/wazuh/assets/69121070/f067304a-8871-4717-93ff-eab35daa5739)

</details>

**Comment by ed108206** [comment]
LGMT
